src/controls/Controls.spec.js

```diff
@@ -1 +1,51 @@
-// Test away!
\ No newline at end of file
+import React from 'react';
+import '@testing-library/jest-dom';
+import {render, fireEvent} from '@testing-library/react';
+import Controls from '../controls/Controls';
+
+describe('buttons text changes to reflect the state the door will be in if clicked', () => {
+    test('button toggle to Open and Unlocked', () =>{
+        const toggleClosed = jest.fn();
+        const toggleLocked =jest.fn();
+
+        const {getByTestId} = render(
+            <Controls
+            data-testid='controls'
+            locked={true}
+            closed={true}
+            toggleClosed = {toggleClosed}
+            toggleLocked = {toggleLocked}
+            />
+        )
+        const locked = getByTestId(/lock-unlock/i);
+        const closed = getByTestId(/open-closed/i);
+        
+        fireEvent.click(locked);
+        fireEvent.click(closed);
+
+        expect(closed.textContent).not.toBe('Closed');
+        expect(locked.textContent).not.toBe('Lock');
+    })
+    test('button toggle to Close and Unlock', () => {
```

Comment:
This test could be shortened.  For instance you would only need to render the dashboard. Get the closed by the getByText then fire the event to close. So it would look more like this `   const { getByText } = render(<Dashboard />
    );
    const closed = getByText(/close gate/i);
    fireEvent.click(closed);
    const locked = getByText(/lock gate/i);
    fireEvent.click(locked);
    const openGate = getByText(/open gate/i);
    expect(openGate).toHaveAttribute('disabled')
`
The nice thing about getting the Attribute is that if the attribute is disabled you can just say disabled without having to get if the text says Locked or Unlocked. But the test is robust to look for the text then that good however you don't need to toggle the locked consistently.